### ISSUE TITLE
bitflags 0.7.1 broke everything.

### ISSUE THREAD

**Issue by samosaara** [Gekkio/imgui-rs#42]
So, I was very undecided If I should make this issue or not since I'm so new to rust and everything and I didn't want to use the issues pages as a StackExchange,  but after some testing and digging around I think I got a reproducible bug here. 
I've gone through so many silly mistakes to get here that feels almost criminal to get to the conclusion, but I'll cut to the chase.
I simply tried to use (as in adding it to Cargo.toml and `cargo build`) your lib through cargo and crates.io (version 0.0.10) and what I got was this:
```
error[E0034]: multiple applicable items in scope
   --> /home/dovahkiin/.cargo/registry/src/github.com-1ecc6299db9ec823/imgui-0.0.10/src/input.rs:98:24
    |
98  |             self.flags.set(ImGuiInputTextFlags_AlwaysInsertMode, value);
    |                        ^^^ multiple `set` found
...
321 | impl_input_intn!(InputInt4, 4, igInputInt4);
    | -------------------------------------------- in this macro invocation
    |
    = note: candidate #1 is defined in an impl for the type `imgui_sys::ImGuiInputTextFlags`
    = note: candidate #2 is defined in an impl for the type `imgui_sys::ImGuiInputTextFlags`

error: aborting due to 140 previous errors
```
And the other 140 were very very similar to that one changing only stuff like `mGuiInputTextFlags_NoHorizontalScroll,` instead of always insert mode, etc. Basically 140 occurrences of the same bug in different 'methods', 'identifiers'??

And after many silly mistakes, I finally got to this next error when trying to build this whole repo (cloning and `cargo build`), error which in turn I think is what is causing or at least has some relation with the first one:

``` 
error: duplicate definitions with name set, #[deny(overlapping_inherent_impls)] on by default
   --> imgui-sys/src/lib.rs:295:5
    |
295 |       pub fn set(&mut self, mask: ImGuiTreeNodeFlags, value: bool) {
    |  _____^ starting here...
296 | |         if value {
297 | |             *self |= mask;
298 | |         } else {
299 | |             *self -= mask;
300 | |         }
301 | |     }
    | |_____^ ...ending here
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #36889 <https://github.com/rust-lang/rust/issues/36889>

error: aborting due to 3 previous errors 
```

And yes, once again the previous 3 errors are the same in different identifiers. I can post the whole logs for both errors if necessary, but I tried it multiple times and they seem easily reproducible.

Environment: 
- OS: Antergos Linux (arch-linux KISS reskin)
- rustc: `1.15.1 (021bd294c 2017-02-08)`
- cargo: `0.16.0-nightly (6e0c18c 2017-01-27)`

**Comment by Gekkio** [comment]
Thanks for reporting this!

This issue is caused by the latest `bitflags` library release (0.7.1), which introduces a new feature in an incompatible way. `imgui-sys` depends on `bitflags` 0.7, which means it automatically uses 0.7.1 based on semver rules. I'll need to fix this incompatibility in `imgui-sys` and bump the minimum `bitflags` version to 0.7.1, because it won't be possible to support both 0.7.0 and 0.7.1 at the same time.

**Comment by BurntSushi** [comment]
@Gekkio What change are you referring to in `bitflags`? It seems like there should have been a semver bump?

**Comment by Gekkio** [comment]
This one: https://github.com/rust-lang-nursery/bitflags/commit/67e4485d20ac79128843fb14abd3e8df742fcfae

If a new function is added to a normal library, that's not a breaking change. But for macros, adding a new function to generated structs can be a breaking change because it can conflict with an existing function with the same name. In this case I already have a `set` function implemented for all of my bitflag structs. The fix is to remove my implementations because they are no longer needed.

**Comment by BurntSushi** [comment]
I filed an issue upstream: https://github.com/rust-lang-nursery/bitflags/issues/59

Thanks for elaborating!

**Comment by Gekkio** [comment]
Thanks!

**Comment by samosaara** [comment]
Glad you were able to pinpoint the bug so fast, I didn't even consider trying to check the libraries versions. And it's dead on. I downgraded it back to 0.7.0 and worked flawlessly. The title feels so irrelevant now. :laughing: waiting eagerly to 0.1.1 as soon as you're able.

**Comment by Gekkio** [comment]
bitflags 0.7.1 was yanked so existing imgui-rs releases should now compile just fine. I also released 0.0.11 with some fixes and it uses bitflags 0.8 as planned :smile:
